pkg/kube/pod.go

```diff
@@ -147,6 +149,14 @@ func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions)
 		pod.SetOwnerReferences(opts.OwnerReferences)
 	}
 
+	if opts.PodSecurityContext != nil {
+		pod.Spec.SecurityContext = opts.PodSecurityContext
+	}
+
+	if opts.ContainerSecurityContext != nil {
+		pod.Spec.Containers[0].SecurityContext = opts.ContainerSecurityContext
+	}
+
```

Comment:
> Hi @TimonOmsk Can we not have pod and container security context as part of `podOverride` itself?

I would prefer to avoid podOverride. Actually I think that the whole concept of podOverride is not a very good idea because there is no control over what exactly we got using override and how it was merged. Some overrides may clash which can lead to unpredictable behaviour. Also, I believe that security is one of basic parameter for Pod, so I think it would be great to set it explicitly.

Reply:
I agree about the complexity that we get with using `podOverride`. But I think it's going to be a bit confusing on how do we want to set the security context. If we really want to get this PR in, I don't have a problem but we should heavily document.
Also, does this impact the `podOverride` field that we specify for some kanister functions?

Reply:
@viveksinghggits can you please point me to those "some kanister functions" which can be affected? 
About docs. Just say what you think I should document and I'll do that :)

Reply:
https://docs.kanister.io/functions.html#kubetask is one example of kanister function where we can pass the pod spec to override the spec of the pod that kanister creates.
My concern is, should something be changed/documented here?
if, just comment in the source code would be sufficient, specifying that because of the complexity of `podOverride` we are accepting the pod and container security context separately.

Reply:
As far as I can see we don't have any mentions about security specific settings for podOverride field in the docs. So for me it means that we can specify somewhere in docs the default behaviour of pods(running as root) and some recommendations how user can set other security related settings.

Reply:
If behaviour of those functions is not changing in anyway in that case I recommend we should just document this in source file with comments. Instead of mentioning it in public docs.